synapse/common.py

```diff
@@ -535,6 +531,14 @@ def spin(genr):
     '''
     collections.deque(genr, 0)
 
+async def aspin(genr):
+    '''
+    Async version of spin
+    '''
+    # FIXME:  is there anything faster?
```

Comment:
probably not. spin's `collections.deque(genr, 0)` takes advantage of cpython's dequeue implementation. I don't know if anything similar exists for async.